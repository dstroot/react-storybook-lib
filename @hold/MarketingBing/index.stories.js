import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import MarketingBing from '../MarketingBing';

// MarketingBing section
storiesOf('MarketingBing', module).add('show MarketingBing', () => (
  <MarketingBing
    index="0"
    lead="Our Commitments"
    text="We are dedicated to creating opportunity for everyone"
    cta="See how we contribute"
    url="/"
  />
));
