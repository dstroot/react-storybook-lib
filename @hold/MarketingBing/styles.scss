/* The image box */
.image-box {
  position: relative;
  height: 425px;
}

/* Place text on the image */
.image-box-text {
  max-width: 650px;
  position: absolute;
  text-align: center;
  top: 20%;
}
@media (min-width: 768px) {
  .image-box-text {
    margin-left: 8.6%;
    max-width: 450px;
    position: absolute;
    text-align: left;
    top: 45%;
  }
}

// darken on rollover
.image-box:hover .overlay {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  height: 100%;
  width: 100%;
  background-color: rgba(0, 142, 215, 0.6);
  transition: 0.5s ease;
}

// images and info
// ----------------
// https://bing.biturl.top/?resolution=1920&format=image&index=0&mkt=en-US
// http://unsplash.it/1200x800
// ----------------
// https://www.google.com/intl/en/about/
// https://www.w3schools.com/css/css3_backgrounds.asp
