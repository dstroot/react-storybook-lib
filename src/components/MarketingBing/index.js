import React from 'react';
import PropTypes from 'prop-types';

const MarketingBing = ({ index, lead, text, cta, url }) => {
  return (
    <a href={url}>
      <div
        className="image-box mt-5"
        style={{
          background: `linear-gradient(to right, rgba(0, 0, 0, 0.8), rgba(0, 0, 0, 0)),
      url('https://bing.biturl.top/?resolution=1920&format=image&index=${index}&mkt=en-US')
        center center no-repeat`,
          backgroundSize: `cover`,
        }}
      >
        <div className="overlay">
          <div className="image-box-text text-white">
            <p className="lead text-uppercase">{lead}</p>
            <h3 data-unite-orphan="">{text}</h3>
            <p className="mt-4">{cta}</p>
          </div>
        </div>
      </div>
    </a>
  );
};

export default MarketingBing;

MarketingBing.propTypes = {
  index: PropTypes.string.isRequired,
  lead: PropTypes.string.isRequired,
  cta: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
