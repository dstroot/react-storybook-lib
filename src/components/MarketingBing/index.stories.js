import React from "react";
import { storiesOf } from "@storybook/react";

// components
import MarketingBing from "../MarketingBing";

// MarketingBing section
storiesOf("MarketingBing", module)
  .addParameters({
    info: {
      inline: true,
      header: false
    }
  })
  .add("show MarketingBing", () => (
    <MarketingBing
      index="0"
      lead="Our Commitments"
      text="We are dedicated to creating opportunity for everyone"
      cta="See how we contribute"
      url="/"
    />
  ));
