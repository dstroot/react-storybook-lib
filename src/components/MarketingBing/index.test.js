import React from 'react';
import renderer from 'react-test-renderer';
import MarketingBing from '../MarketingBing';

describe('MarketingBing', () => {
  it('it should render', () => {
    const component = renderer.create(
      <MarketingBing
        index="1"
        lead="Our Commitments"
        text="We are dedicated to creating opportunity for everyone"
        cta="How we contribute"
        url="/"
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
