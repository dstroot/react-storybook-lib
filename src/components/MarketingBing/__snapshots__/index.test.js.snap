// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketingBing it should render 1`] = `
<a
  href="/"
>
  <div
    className="image-box mt-5"
    style={
      Object {
        "background": "linear-gradient(to right, rgba(0, 0, 0, 0.8), rgba(0, 0, 0, 0)),
      url('https://bing.biturl.top/?resolution=1920&format=image&index=1&mkt=en-US')
        center center no-repeat",
        "backgroundSize": "cover",
      }
    }
  >
    <div
      className="overlay"
    >
      <div
        className="image-box-text text-white"
      >
        <p
          className="lead text-uppercase"
        >
          Our Commitments
        </p>
        <h3
          data-unite-orphan=""
        >
          We are dedicated to creating opportunity for everyone
        </h3>
        <p
          className="mt-4"
        >
          How we contribute
        </p>
      </div>
    </div>
  </div>
</a>
`;
